src/MessagePack.UnityClient/Assets/Scripts/MessagePack/Formatters/StandardClassLibraryFormatter.cs

```diff
@@ -108,13 +108,76 @@ private DecimalFormatter()
 
         public void Serialize(ref MessagePackWriter writer, decimal value, MessagePackSerializerOptions options)
         {
-            writer.Write(value.ToString(CultureInfo.InvariantCulture));
-            return;
+            var dest = writer.GetSpan(MessagePackRange.MaxFixStringLength);
+            if (System.Buffers.Text.Utf8Formatter.TryFormat(value, dest.Slice(1), out var written))
+            {
+                // write string header
+                dest[0] = (byte)(MessagePackCode.MinFixStr | written);
+                writer.Advance(written + 1);
+            }
+            else
+            {
+                writer.Write(value.ToString(CultureInfo.InvariantCulture));
+            }
         }
 
         public decimal Deserialize(ref MessagePackReader reader, MessagePackSerializerOptions options)
         {
-            return decimal.Parse(reader.ReadString(), CultureInfo.InvariantCulture);
+            if (!(reader.ReadStringSequence() is ReadOnlySequence<byte> sequence))
+            {
+                throw new MessagePackSerializationException(string.Format("Unexpected msgpack code {0} ({1}) encountered.", MessagePackCode.Nil, MessagePackCode.ToFormatName(MessagePackCode.Nil)));
+            }
+
+            if (sequence.IsSingleSegment)
+            {
+                if (System.Buffers.Text.Utf8Parser.TryParse(sequence.First.Span, out decimal result, out _))
```

Comment:
Consider throwing if `bytesConsumed < sequence.Length` since that's unexpected.

Reply:
OK